aspnet/client-side/bundling-and-minification.rst

```diff
@@ -1,8 +1,299 @@
-.. include:: /../common/stub-topic.txt
+Bundling and Minification
+=========================
 
-|stub-icon| Bundling and Minification
-=====================================
+By `Erik Reitan`_ 
 
-.. include:: /../common/stub-notice.txt
+Bundling and minification are two techniques you can use in ASP.NET 5 to improve page load performance for your web application. Bundling makes it easy to combine or bundle multiple files into a single file. Minification performs a variety of different code optimizations to scripts and CSS, which results in smaller file sizes. Bundling and minification improves load time performance by reducing the number of requests to the server and reducing the size of the requested assets (such as CSS and JavaScript files.)
 
-.. _issue: https://github.com/aspnet/Docs/issues/101
+This article explains and shows how to use the bundling and minification features included with the ASP.NET 5 Web Application template in `Visual Studio 2015 <http://go.microsoft.com/fwlink/?LinkId=517106>`_. 
+
+Overview
+--------
+Because bundling combines multiple files into a single file, it reduces the number of requests to the server that is required to retrieve and display a web asset, such as a web page. You can create CSS, JavaScript and other bundles. Fewer files, means fewer HTTP requests from your browser to the server or from the service providing your application. This results in improved first page load performance.
+
+Minification performs a variety of different code optimizations to reduce the size of requested assets (such as CSS, image, JavaScript files). Common results of minification include removing unnecessary white space and comments, and shortening variable names to one character. 
+
+Consider the following JavaScript function:
+
+.. code-block:: javascript
+
+	AddAltToImg = function (imageTagAndImageID, imageContext) {
+		///<signature>
+		///<summary> Adds an alt tab to the image
+		// </summary>
+		//<param name="imgElement" type="String">The image selector.</param>
+		//<param name="ContextForImage" type="String">The image context.</param>
+		///</signature>
+		var imageElement = $(imageTagAndImageID, imageContext);
+		imageElement.attr('alt', imageElement.attr('id').replace(/ID/, ''));
+	}
+
+After minification, the function is reduced to the following:
+
+.. code-block:: javascript
+
+	AddAltToImg=function(t,a){var r=$(t,a);r.attr("alt",r.attr("id").replace(/ID/,""))};
+	
+In addition to removing the comments and unnecessary whitespace, the following parameters and variable names were renamed (shortened) as follows:
+
+==================  =======  
+Original            Renamed      
+==================  =======  
+imageTagAndImageID  n  
+imageContext        t  
+imageElement        i   
+==================  =======  
+
+ASP.NET Web Application Template 
```

Comment:
I feel like we're missing a section here on the high level flow before jumping into the implementation in the templates. Remember, most ASP.NET customers today are doing runtime bundling and minification using ASP.NET Web Optimization. The idea of doing design time bundling and minification will be new to many folks, so I think we should introduce that concept up front (i.e. you build your minified files up front and then deploy them with your app as static files). It would be good to highlight that advantages of this approach (fewer moving parts at runtime, reduced server load) and also mention the limitations (addition build complexity, only works with static content).